map/framework.cpp

```diff
@@ -1520,21 +1386,18 @@ void Framework::InitUGC()
   m_ugcApi = make_unique<ugc::Api>(m_model.GetIndex());
 }
 
-void Framework::InitSearchEngine()
+void Framework::InitSearchAPI()
 {
-  ASSERT(!m_searchEngine.get(), ("InitSearchEngine() must be called only once."));
+  ASSERT(!m_searchAPI.get(), ("InitSearchAPI() must be called only once."));
   ASSERT(m_infoGetter.get(), ());
   try
   {
-    search::Engine::Params params;
-    params.m_locale = languages::GetCurrentOrig();
-    params.m_numThreads = 1;
-    m_searchEngine = make_unique<search::Engine>(const_cast<Index &>(m_model.GetIndex()),
-                                                 GetDefaultCategories(), *m_infoGetter, params);
+    m_searchAPI = make_unique<SearchAPI>(const_cast<Index &>(m_model.GetIndex()), m_storage,
```

Comment:
there is a non-const method ```FeaturesFetcher::GetIndex``` so you don't need to do ```const_cast```

Reply:
Could you please clarify how FeaturesFeatcher is related here? m_model is a FeatureVecModel.